### PR TITLE
elb_application_lb: test a rule based on http-header

### PR DESCRIPTION
**Pull request by goneri**
This should initially fail and be fixed the day https://github.com/ansible-collections/amazon.aws/pull/188
is merged.

See: https://github.com/ansible-collections/amazon.aws/pull/187
See: https://github.com/ansible-collections/community.aws/issues/117

### CODE REVIEW

**Comment by goneri** [comment]
On hold until https://github.com/ansible-collections/community.aws/pull/348 is merged.

**Comment by goneri** [comment]
This feature depends on boto3 1.16.57 which is not availbe in our CI env. https://github.com/ansible-collections/amazon.aws/pull/188#issuecomment-763099730

**Comment by jillr** [comment]
@goneri 1.16.57 is very new, it's only been out for a few weeks.  It's unlikely many of our users will have it yet in their environments.

Rather than requiring this version of boto3 in the mod_util we should check in the affected module(s) for the boto3 version when the feature that needs this version is used, like we do here:
https://github.com/ansible-collections/community.aws/blob/main/plugins/modules/iam_role.py#L641-L645

Then the integration tests would setup a virtualenv with that version of boto3 when testing that feature: https://github.com/ansible-collections/community.aws/blob/main/tests/integration/targets/elb_application_lb/tasks/main.yml

File: tests/integration/targets/elb_application_lb/tasks/test_multiple_actions.yml [hunk 1]
```diff
@@ -175,6 +175,22 @@
             - Type: fixed-response
               FixedResponseConfig: "{{ FixedResponseActionConfig }}"
           Rules:
+            - Conditions:
+                - Field: http-header
+                  HttpHeaderConfig:
+                    Field: 'User-Agent'
```
Comment (tremble) [inline tests/integration/targets/elb_application_lb/tasks/test_multiple_actions.yml:181]:
```suggestion
                    HttpHeaderName: 'User-Agent'
```

File: tests/integration/targets/elb_application_lb/tasks/test_multiple_actions.yml [hunk 2]
```diff
@@ -175,6 +175,22 @@
...
+                    Values: ['*Trident/7:0*rv:*']
+                - Field: http-header
+                  HttpHeaderConfig:
+                    Field: 'X-Something'
```
Comment (tremble) [inline tests/integration/targets/elb_application_lb/tasks/test_multiple_actions.yml:185]:
```suggestion
                    HttpHeaderName: 'X-Something'
```

**Review by tremble — CHANGES_REQUESTED** [review]
Some examples of adding the new version of boto to environments can be found in:
https://github.com/ansible-collections/community.aws/pull/390

Note: I would generally recommend putting the dependency on the version of botocore rather than boto3.  (That's where the actual test is happening)

to find the exact version of botocore you can do the following:

```
git clone https://github.com/boto/botocore.git
cd botocore
git blame botocore/data/elbv2/2015-12-01/service-2.json
```

Search for the API parameter you're trying to set (in this case HttpHeaderConfig):

```
4667b46391 (aws-sdk-python-automation 2019-06-24 20:15:37 +0000 2657)         "HttpHeaderConfig":{
4667b46391 (aws-sdk-python-automation 2019-06-24 20:15:37 +0000 2658)           "shape":"HttpHeaderConditionConfig",
4667b46391 (aws-sdk-python-automation 2019-06-24 20:15:37 +0000 2659)           "documentation":"<p>Information for an HTTP header condition. Specify only when <code>Field</code> is <code>http-header</code>.</p>"
4667b46391 (aws-sdk-python-automation 2019-06-24 20:15:37 +0000 2660)         },
```

The commit hash at the start of the line then tells you when the option was added to botocore

```
git tag --contains 4667b46391
```

This will then spit out a list of all botocore versions with support for that feature
the earliest being 1.12.176 ( Tue Jun 25 19:29:47 2019 ) which is much more reasonable to expect people to have installed than one from 2 weeks ago.

Digging a little further it looks like you're passing the wrong parameters for HttpHeaderConfig.  See also https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/elbv2.html#ElasticLoadBalancingv2.Client.create_rule

**Review by tremble — APPROVED** [review]

**Pull request merged**